modules/_labs/color-picker/react/lib/parts/SwatchBook.tsx

```diff
@@ -37,6 +41,10 @@ const SwatchContainer = styled('div')<SwatchContainerProps>(
       ...focusRing({separation: 2}),
     },
   },
+  ({index}) => ({
+    '-ms-grid-column': (index % numberOfSwatchColumns) + 1 + '',
+    '-ms-grid-row': Math.ceil((index + 1) / numberOfSwatchColumns) + '',
+  }),
```

Comment:
Hmm. Lack of autoplacement of grid items makes me wonder why we should use CSS grid for this layout instead of a single `display: flex` or a `display: flex` per row.

Emotion does a lot of stuff behinds the scenes that creates performance issues and this is one of them. For every cell (row/column combination), a new style rule is injected into the DOM containing _all_ styles in this block. Not just a style object with the properties defined here, but all of them. In this example, that's 22 properties - 20 of which are repeats. In this case, we have 24 unique indexes, so 24 * 20 = 480 style rules for a 24 boxes that only have a color difference.

We're doing the same thing for each swatch as well... In that case it's 16 style rules for each unique color where only the background color changes. That's 912 extra style rules that aren't adding any extra value - just space.

Reply:
Do you advise I remove the grid layout, or enhance the logic to be more performant, or leave as is for this PR?

Reply:
At the risk of missing something glaring...

Because the popup width is fixed, could we just make the `SwatchContainer` `display: inline-flex` or make the `Container` a flexbox that specifies `flex-wrap: wrap`. There would be no maths required. It's a little bit of a naive implementation but it should be easier to reason about and not have a perf hit.

Reply:
👍